### DISCUSSION TITLE
Laravel 11 Support

### DISCUSSION THREAD

**Discussion by vonec** [spatie/laravel-responsecache#465] [General]
Is laravel 11 supported ?

**Comment by freekmurze** [comment]
Yes

**Reply by Fot0n** [reply]
The location and order of adding middleware to Laravel 11 have changed. The documentation in this part is outdated.

**Comment by vonec** [comment]
spatie/laravel-responsecache 7.5.0 requires nesbot/carbon ^2.63 -> found nesbot/carbon[2.63.0, ..., 2.x-dev] but the package is fixed to 3.1.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

carbon version mismatch

**Reply by freekmurze** [reply]
We now allow Carbon v3 👍

**Reply by vonec** [reply]
We met at Laracon India last year, and I'm filled with immense gratitude for all the wonderful work you've accomplished. Unfortunately, I won't be able to attend this year, but I hope you have an incredible time at Laracon India. Happy Holi

**Reply by freekmurze** [reply]
Thank you 🙂

**Comment by francoism90** [comment]
@freekmurze I think something else isn't working as expecting after upgrading to Laravel 11.

The problem are the cache-keys, I'm keep getting they are unsupported by the driver, even when I use `redis`. Is this a known problem?

**Comment by freekmurze** [comment]
It seems like the tests are passing. Could you PR a non working test?

**Reply by francoism90** [reply]
@freekmurze I'll try, but unfortunately it seems Laravel completely removed the section about using cache tags, even when still on Laravel 10.

I don't know if it still supported by upstream or shouldn't be used at all anymore. I remember something about being to complex to maintain, so don't know if it's still advisable?

I've asked to put the section back, but unfortunately didn't get any response. I'm afraid it just isn't supported anymore.

**Reply by francoism90** [reply]
@freekmurze So it does work, but I didn't set the cache profile to redis.. sorry about that!

To make sure the stale-tags are being cleared, I would suggest adding this to `routes/console.php` for L11:
```php
<?php
use Illuminate\Cache\Console\PruneStaleTagsCommand;

Schedule::command(PruneStaleTagsCommand::class)
    ->withoutOverlapping(600)
    ->hourly()
    ->runInBackground();
```

**Comment by RasmusLindstrom** [comment]
Trying to use with Laravel 11 (no default Kernel file) is there an example of how to use with the bootstrap/app.php file?

**Reply by FT-Riddhi** [reply]
yes.  use the bootstrap/app.php file

**Reply by francoism90** [reply]
@RasmusLindstrom Something like this:

```
// bootstrap.app.php

->withMiddleware(function (Middleware $middleware) {
        $middleware->throttleWithRedis();
        $middleware->statefulApi();
        $middleware->redirectGuestsTo('/login');

        $middleware->web(append: [
            HandleInertiaRequests::class,
            AddLinkHeadersForPreloadedAssets::class,
        ]);

        $middleware->alias([
            'abilities' => \Laravel\Sanctum\Http\Middleware\CheckAbilities::class,
            'ability' => \Laravel\Sanctum\Http\Middleware\CheckForAnyAbility::class,
            'cache' => \Foundation\Http\Middlewares\SetCacheHeaders::class,
            'cache_model' => \Support\ResponseCache\Middlewares\CacheModelResponse::class,
            'cache_response' => \Spatie\ResponseCache\Middlewares\CacheResponse::class, // this one
            'private' => \Foundation\Http\Middlewares\EnsureRequestHasPrivateSubnet::class,
            'permission' => \Spatie\Permission\Middleware\PermissionMiddleware::class,
            'role_or_permission' => \Spatie\Permission\Middleware\RoleOrPermissionMiddleware::class,
            'role' => \Spatie\Permission\Middleware\RoleMiddleware::class,
        ]);
    })
```
You basically add this as an `alias`. In your controller you can use the middleware helper to call `cache_response`.
